activerecord/lib/active_record/collection_cache_key.rb

```diff
@@ -0,0 +1,27 @@
+module ActiveRecord
+  module CollectionCacheKey
+
+    # Generates a cache key for the records in the given collection.
+    # See <tt>ActiveRecord::Relation#cache_key</tt> for details.
+    def collection_cache_key(collection = all, timestamp_column = :updated_at)
+      query_signature = Digest::MD5.hexdigest(collection.to_sql)
+      key = "#{collection.model_name.cache_key}/query-#{query_signature}"
+
+      size, timestamp = if collection.loaded?
+        [collection.size, collection.collect(&timestamp_column).compact.max]
```

Comment:
Do you think this would be better written as: `collection.max_by(&timestamp_column).public_send(timestamp_column)`